plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/controls/CustomFormEditor.java

```diff
@@ -301,17 +305,48 @@ public void widgetSelected(SelectionEvent e) {
         } else if (isTextPropertyType(propType)) {
-            if (property instanceof ObjectPropertyDescriptor && ((ObjectPropertyDescriptor) property).getLength() == PropertyLength.MULTILINE) {
+            if (property instanceof ObjectPropertyDescriptor && property.getLength() == PropertyLength.MULTILINE) {
                 Label label = UIUtils.createControlLabel(parent, propertyDisplayName);
                 label.setLayoutData(new GridData(GridData.VERTICAL_ALIGN_BEGINNING));
-                Text editor = new Text(parent, SWT.BORDER | SWT.MULTI | SWT.V_SCROLL | SWT.WRAP | (readOnly ? SWT.READ_ONLY : SWT.NONE));
-
+                Composite composite = UIUtils.createComposite(parent, 2);
+                Text editor = new Text(composite, SWT.MULTI | SWT.WRAP | SWT.BORDER | (readOnly ? SWT.READ_ONLY : SWT.NONE));
+                editor.setLayoutData(new GridData(GridData.FILL_BOTH));
                 editor.setText(objectValueToString(value));
-                GridData gd = new GridData(GridData.FILL_HORIZONTAL);
-                // Make multiline editor at least two lines height
-                gd.heightHint = (UIUtils.getTextHeight(editor) + editor.getBorderWidth()) * 2;
-                editor.setLayoutData(gd);
-                return editor;
+
+                int textHeight = UIUtils.getTextHeight(editor) * 3;
+                GridData gdEditor = new GridData(SWT.FILL, SWT.TOP, true, false);
+                gdEditor.heightHint = textHeight;
+                composite.setLayoutData(gdEditor);
+
+                Button editButton = new Button(composite, SWT.FLAT | SWT.PUSH);
+                editButton.setImage(DBeaverIcons.getImage(UIIcon.DOTS_BUTTON));
+                GridData gd = new GridData(SWT.END, SWT.NONE, false, true);
+                editButton.setLayoutData(gd);
+                editButton.addSelectionListener(new SelectionAdapter() {
+                    @Override
+                    public void widgetSelected(SelectionEvent e) {
+                        if (readOnly) {
+                            EditTextDialog.showText(
+                                editor.getShell(),
+                                UIMessages.edit_text_dialog_title_edit_value,
+                                editor.getText()
+                            );
+                        } else {
+                            String newValue = EditTextDialog.editText(
+                                editor.getShell(),
+                                UIMessages.edit_text_dialog_title_edit_value,
+                                editor.getText()
+                            );
+                            if (newValue != null) {
+                                editor.setText(newValue);
+                            }
+                        }
+                    }
+                });
+                GridData gdComposite = new GridData(SWT.FILL, SWT.TOP, true, false);
+                gdComposite.heightHint = textHeight + 6;
```

Comment:
Maybe I would make the button height smaller, but I am not sure how exactly it would look (if it would be better)

Reply:
![image](https://github.com/user-attachments/assets/21389e5d-f5a9-405a-bdfb-ef98c9718794)